Description: Add casts to silence a compiler warning
Origin: upstream, git://git.ghostscript.com/jbig2dec.git;a=commit;h=d8294b2
Author: Sebastian Rasmussen <sebras@gmail.com>
Forwarded: yes
Last-Update: 2021-02-02
---
This patch header follows DEP-3: https://dep.debian.net/deps/dep3/
--- a/jbig2_image.c
+++ b/jbig2_image.c
@@ -347,8 +347,8 @@
     if (src == NULL)
         return 0;
 
-    if ((UINT32_MAX - src->width  < (x > 0 ? x : -x)) ||
-        (UINT32_MAX - src->height < (y > 0 ? y : -y)))
+    if ((UINT32_MAX - src->width  < (uint32_t) (x > 0 ? x : -x)) ||
+        (UINT32_MAX - src->height < (uint32_t) (y > 0 ? y : -y)))
     {
 #ifdef JBIG2_DEBUG
         jbig2_error(ctx, JBIG2_SEVERITY_DEBUG, JBIG2_UNKNOWN_SEGMENT_NUMBER, "overflow in compose_image");
